src/packages/version-check/tests/version-check.test.ts

```diff
@@ -0,0 +1,689 @@
+        true,
+        "lastVersionLogged was not successfully set after logging version message"
+      );
+    });
+    it("only logs the latestVersion one time", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+
+      vc.log();
+
+      assert.strictEqual(
+        message.length > 0,
+        true,
+        "Log did not log the first time for this version"
+      );
+      message = "";
+      vc.log();
+
+      assert.strictEqual(message, "", "logged the same version message twice");
+    });
+
+    it("message contains the upgradeType", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      const upgradeType = semverUpgradeType(
+        vc._currentVersion,
+        vc._config.latestVersion
+      );
+
+      assert.strictEqual(
+        message.indexOf(upgradeType) >= 0,
+        true,
+        "Message does not contain the upgradeType"
+      );
+    });
+
+    it("message contains the packageName", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.strictEqual(
+        message.indexOf(vc._config.packageName) >= 0,
+        true,
+        "Message does not contain the upgradeType"
+      );
+    });
+    it("message contains the currentVersion", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.strictEqual(
+        message.indexOf(vc._currentVersion) >= 0,
+        true,
+        "Message does not contain the upgradeType"
+      );
+    });
+    it("message contains the latestVersion", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.equal(
+        message.indexOf(vc._config.latestVersion) >= 0,
+        true,
+        "Message does not contain the upgradeType"
+      );
+    });
+    it("process.stdout.columns === -1", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      process.stdout.columns = -1;
+
+      vc.log();
+
+      assert.equal(message.length > 0, true, "Message did not log");
+    });
+    it("process.stdout.columns === null", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      process.stdout.columns = 0;
+
+      vc.log();
+
+      assert.equal(message.length > 0, true, "Message did not log");
+    });
+  });
+
+  describe("getLatestVersion/_fetchLatest", () => {
+    let api;
+    const apiResponse = "1.0.0";
+    const apiSettings = {
+      port: 4000
+    };
+
+    beforeEach(() => {
+      api = http2.createServer();
+      api.on("error", err => console.error(err));
+
+      api.on("stream", (stream, headers) => {
+        const path = headers[":path"];
+        const method = headers[":method"];
+
+        if (method !== "GET") {
+          stream.respond({
+            ":status": 404
+          });
+          stream.end();
+        }
+        switch (path) {
+          case "/?name=slow":
```

Comment:
@satyajeetkolhapure if you are still looking for a way to test your connection on exponential backoff this is a way to start a server that will accept a response but never respond.

Reply:
Awesome. I will use this.